kcidb/monitor/subscriptions/linux_stable_rc.py

```diff
@@ -19,6 +19,22 @@ def match_revision(revision):
     if revision.builds_valid is None:
         return ()
 
+    # Only take into account results from maestro and broonie as of now
+    checkouts_copy = revision.checkouts.copy()
+    for checkout in checkouts_copy:
+        if checkout.origin not in ("maestro", "broonie"):
+            revision.checkouts.remove(checkout)
```

Comment:
Hi @spbnick 
To report results only from `maestro` and `broonie`, we need to filter out the data accordingly.
Filter for `revision.checkouts` isn't working where builds and tests are working fine.
Could you please check what's wrong with this?
I also observed `self.checkouts` in `Revision` class displays different memory address for checkout objects than `revision.checkouts` here.

Reply:
OK, first of  all, please  don't remove anything from the objects. They're cached, and they will be reused for other subscriptions. Treat them as read-only. I don't think I ever implemented proper copying support, so don't rely on that. Finally, even if you copy and modify them somehow, the templates won't see those modifications, because they're supplied with the original objects provided to your subscription's match function, not your copies.

You'll have to filter the objects on the fly as you're iterating through them in your subscription function, as well as in your templates. Thankfully both Python and Jinja2 (to some extent) are good at that.

Otherwise, we would need to implement advanced filtering, where you could get the same object, but with only specified origin's data. I've been planning that for a while, but so far haven't really needed it. Perhaps we'll need to do that soon. For now, please try ad-hoc filtering and see how hard/easy it is.

> I also observed self.checkouts in Revision class displays different memory address for checkout objects than revision.checkouts here.

I would have to see exact code reproducing this to tell you what's going on.

Reply:
Okay. Thank you. I have updated the code accordingly.